lib/planner/search-algorithms/offline-replanning.js

```diff
@@ -307,25 +374,42 @@ module.exports = offlineReplanning = {
     let next = yield;
     let backtrackPlan = null;
     let newSavedPlans = null;
-
+    let removed = [];
     while (index >= 0) {
-      backtrackPlan = yield offlineReplanning.emitAsync('searchNode.clone', offlineReplanning._savedPlans[index], next);
+      backtrackPlan = yield offlineReplanning.emitAsync('searchNode.clone',
+          offlineReplanning._savedPlans[index], next); // TODO: Potentially no need to clone—future improvement?
       let actions = new Set(backtrackPlan.actions);
 
-      actions.delete(offlineReplanning._savedPlans[index + 1].lastAction);
+      removed.push(offlineReplanning._savedPlans[index + 1].lastAction);
+      actions.delete(removed[removed.length - 1]);
 
       if (actions.size > 0) {
         newSavedPlans = offlineReplanning._savedPlans.slice(0, index + 1);
         backtrackPlan.actions = actions;
         offlineReplanning._savedPlans[index].actions = actions;
-        return callback(null, {backtrackPlan, newSavedPlans});
+        return callback(null, {backtrackPlan, newSavedPlans, removed});
       }
 
       index--;
     }
 
     return callback(null, {backtrackPlan: null, newSavedPlans: null});
   },
+  _comparePlan(plan) {
+    const hash = crypto.createHash('sha256');
+    let planString = JSON.stringify(plan.path);
+    const planHashDigest = hash.update(planString).digest('base64');
+
+    return offlineReplanning._planHashes.get(planHashDigest);
+  },
+  _random(min, max) {
+    max = max ? max : 1;
+    min = min ? min : 0;
+    offlineReplanning._randomSeed = (offlineReplanning._randomSeed * 9301 + 49297) % 233280;
```

Comment:
Do `9301` and the other literals have any significance behind them?

Reply:
I saw it in an article discussing ways to do a seeded random number generator, but it seems like it was pulled from a book's big list of possible values to make a "somewhat uniform" distribution for random numbers (seem to be in many, here's one: https://books.google.com/books?id=cyQ_pIMYPEUC&pg=PA218&dq=9301+49297+233280&hl=en&newbks=1&newbks_redir=0&sa=X&ved=2ahUKEwip6f2u4drlAhUox1kKHamgC9AQuwUwAXoECAQQCA#v=onepage&q=9301%2049297%20233280&f=false ).

That is to say: the random values here should not be considered cryptographically secure. We just need some variability, and this seemed to work well enough.

Reply:
Solid